tests/unit/modules/test_asset.py

```diff
@@ -13,25 +14,242 @@
+        assert asset._do_builds_differ(
+            [{
+                "sha512": "a",
+                "url": "a",
+                "filters": ["a", "b", "c"],
+            }],
+            [{
+                "sha512": "a",
+                "url": "a",
+                "filters": ["c", "a", "b"],
+            }],
+        ) is False
+
+    def test_equal_with_none_values(self):
+        assert asset._do_builds_differ(
+            [{
+                "sha512": "a",
+                "url": "a",
+                "filters": None
+            }],
+            [{
+                "sha512": "a",
+                "url": "a",
+                "headers": None,
+            }],
+        ) is False
+
+    def test_equal(self):
+        assert asset._do_builds_differ(
+            [
+                {
+                    "url": "a",
+                    "sha512": "a",
+                    "filters": ["a", "b", "c"],
+                    "headers": {
+                        "foo": "bar",
+                        "bar": "foo",
+                    },
+                },
+                {
+                    "url": "b",
+                    "sha512": "b",
+                    "filters": ["d", "e", "f"],
+                    "headers": {
+                        "abc": "abc",
+                        "def": "def",
+                    },
+                },
+            ],
+            [
+                {
+                    "url": "b",
+                    "sha512": "b",
+                    "filters": ["f", "e", "d"],
+                    "headers": {
+                        "def": "def",
+                        "abc": "abc",
+                    },
+                },
+                {
+                    "url": "a",
+                    "sha512": "a",
+                    "filters": ["b", "c", "a"],
+                    "headers": {
+                        "bar": "foo",
+                        "foo": "bar",
+                    },
+                },
+            ],
+        ) is False
+
+    def test_different_lengths(self):
+        assert asset._do_builds_differ(
+            [{"sha512": "a", "url": "a"}],
+            [
+                {"sha512": "a", "url": "a"},
+                {"sha512": "b", "url": "b"}
+            ],
+        ) is True
+
+    def test_different(self):
+        assert asset._do_builds_differ(
+            [
+                {
+                    "url": "a",
+                    "sha512": "a",
+                    "filters": ["a", "b", "c"],
+                    "headers": {
+                        "foo": "bar",
+                    },
+                }
+            ],
+            [
+                {
+                    "url": "b",
+                    "sha512": "b",
+                    "filters": ["a", "b", "c"],
+                    "headers": {
+                        "abc": "abc",
+                    },
+                },
+            ],
+        ) is True
```

Comment:
We should move this test and add rename it to `test_different_build_content` to give us a hint as to where we should look for changes.